ci/snap/bootstrap/snapcraft.yaml

```diff
@@ -155,9 +163,9 @@ parts:
 
   ubuntu-bootstrap:
     after: [flutter-git]
-    source: .
-    source-type: git
     plugin: nil
+    source: https://github.com/canonical/ubuntu-desktop-provision.git
```

Comment:
Hm, I've only ever seen `source: .` in snapcraft.yamls - I'd double check with someone more experienced if this is a good thing to do. Personally I'd find it rather confusing since I'd expect `snapcraft` to build from my local repo

Reply:
There isn't a better/worse approach both as `source: .` and `source: <url>` are supported references; however they behave differently and so we need to consider that

With `source: .` + local dev, a whack of artefacts make their way into the container and that leads to non-deterministic builds.  We might want this because we iterate on local changes. But critically, we know how to clear artefacts to support local iterations. On my first pass of this PR, I build a script to do this something along the lines of:

```bash
$ flutter clean # needed because the cache has some paths which include $USER
$ snapcraft clean # potentially needed depending on how previous builds went / what changes you made
$ rm -rf packages/subiquity_client/subiquity/.subiquity #  snapcraft fails if artefact is left behind from a local run
$ rm -rf snap # might be needed if the last time you ran it was against the other snap artefact, but also this folder is required for the snapcraft command to run against the right container
```

That's a lot of nuance and I don't think it's worth the hassle, especially because once you know all the nuances its straightforward to run local builds. So, my thinking turned towards the public. Ie someone who might clone the repo, modify locally, and then try and build the snap using `source: .`. In this instance they're going to quickly run into issues and then they'll ask for help or spend a chunk of time tracing through everything to figure out whats going on. I think this group is who we should target with the repos default behaviour because its straightforward to modify the snapcraft.yaml to pull in local changes.

If you'd prefer `source: .` happy to go with that too!

Reply:
I also noticed we don't have any workflows that build the snaps in the github CI now (they used to run on the bootstrap/init-snap branches). If we want to add them back for QA testing on PRs, `source: https://github...` wouldn't work as intended.

Reply:
That's a good pt 🤔  as we defo want to improve our pipeline testing. Okay how about this:

1. Revert to `source: .`
2. I'll update `CONTRIBUTING.md` to explain the nuance around building locally and provide some guidance on how/where things can go wrong

Reply:
Sounds good :+1: